ui/redux/selectors/search.js

```diff
@@ -60,34 +61,54 @@ export const makeSelectHasReachedMaxResultsLength = (query: string): ((state: St
 
 export const makeSelectRecommendedContentForUri = (uri: string) =>
   createSelector(
-    makeSelectClaimForUri(uri),
+    selectHistory,
+    selectClaimsByUri,
     selectSearchResultByQuery,
     makeSelectClaimIsNsfw(uri),
-    (claim, searchUrisByQuery, isMature) => {
+    (history, claimsByUri, searchUrisByQuery, isMature) => {
+      const claim = claimsByUri[uri];
+
+      if (!claim) return;
+
       let recommendedContent;
-      if (claim) {
-        // always grab full URL - this can change once search returns canonical
-        const currentUri = buildURI({ streamClaimId: claim.claim_id, streamName: claim.name });
+      // always grab the claimId - this value won't change for filtering
+      const currentClaimId = claim.claim_id;
 
-        const { title } = claim.value;
+      const { title } = claim.value;
 
-        if (!title) {
-          return;
-        }
+      if (!title) return;
 
-        const options: {
-          related_to?: string,
-          nsfw?: boolean,
-          isBackgroundSearch?: boolean,
-        } = { related_to: claim.claim_id, isBackgroundSearch: true };
+      const options: {
+        related_to?: string,
+        nsfw?: boolean,
+        isBackgroundSearch?: boolean,
+      } = { related_to: claim.claim_id, nsfw: isMature, isBackgroundSearch: true };
 
-        options['nsfw'] = isMature;
-        const searchQuery = getSearchQueryString(title.replace(/\//, ' '), options);
-        const normalizedSearchQuery = createNormalizedSearchKey(searchQuery);
+      const searchQuery = getSearchQueryString(title.replace(/\//, ' '), options);
+      const normalizedSearchQuery = createNormalizedSearchKey(searchQuery);
 
-        let searchResult = searchUrisByQuery[normalizedSearchQuery];
-        if (searchResult) {
-          recommendedContent = searchResult['uris'].filter((searchUri) => searchUri !== currentUri);
+      let searchResult = searchUrisByQuery[normalizedSearchQuery];
+      if (searchResult) {
+        recommendedContent = searchResult['uris'].filter((searchUri) => {
+          const { claimId: searchId } = parseURI(searchUri);
```

Comment:
Good to test/examine whether there's any reason parseURI could fail - we often wrap it in try-catch.